### PR TITLE
Imp script fixes

### PR DESCRIPTION
**Pull request by teodanciu**
# Description

Miscellaneous fixes and improvements related to plutus scripts support in Imp tests: 
* tolerating  non-spending  purpose
* fixing scriptWits fixup, such that both reference and non-reference scripts are supported (by not providing what already is provided) 
* running Alonzo Imp tests in babbage and conway eras  (failing without the previous fix) 
 

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

File: eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs [hunk 1]
```diff
@@ -679,31 +678,6 @@ spec =
...
-          ]
-      it "on non-existant gov-actions" $ do
-        khDRep <- setupSingleDRep
-        gaidConstitutionProp <- submitInitConstitutionGovAction
-        let voter = DRepVoter $ KeyHashObj khDRep
-            dummyGaid = gaidConstitutionProp {gaidGovActionIx = GovActionIx 99} -- non-existant `GovActionId`
-        submitFailingVote
-          voter
-          dummyGaid
-          [injectFailure $ GovActionsDoNotExist $ dummyGaid NE.:| []]
```
Comment (Soupstraw) [inline eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs]: Why are these tests removed?

Comment (teodanciu) [reply eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs]: They are duplicated just below, in the describe "Voting" test

Comment (Soupstraw) [reply eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/GovSpec.hs]: Oh, right

**Review by Soupstraw — APPROVED** [review]
LGTM

**Review by lehins — APPROVED** [review]
:+1:

**Pull request merged**